### PR TITLE
add config options for disabling validators

### PR DESCRIPTION
**Pull request by micheller**

### CODE REVIEW

File: flake8_fine_pytest/watchers/usefixtures.py [hunk 1]
```diff
@@ -4,6 +4,12 @@
 from flake8_fine_pytest.watchers.base import BaseWatcher
 
 
+def is_static_method(node: ast.FunctionDef) -> bool:
```
Comment (mcproger) [inline flake8_fine_pytest/watchers/usefixtures.py:7]: Let's move it to `ast_helpers` module.

File: flake8_fine_pytest/watchers/usefixtures.py [hunk 2]
```diff
@@ -14,24 +20,62 @@ def run(self) -> None:
         if self._should_run():
             self._validate_usefixtures_used_where_possible(self.tree)
 
-    def _validate_usefixtures_used_where_possible(self, tree: ast.AST) -> None:
-        for node in ast.walk(tree):
-            if not self._should_check_node(node):
-                continue
+    def _should_run(self) -> bool:
+        return self.options.force_usefixtures and super()._should_run()
```
Comment (mcproger) [inline flake8_fine_pytest/watchers/usefixtures.py:24]: I understand correctly that we have to do such a check for all settings? How about to make a class attribute and take the validation to the base class via `getattr`?

File: flake8_fine_pytest/watchers/usefixtures.py [hunk 3]
```diff
@@ -14,24 +20,62 @@ def run(self) -> None:
...
+
+        As pytest only discovers first-level-of-nesting tests, we only yield
+        top-level function definitions and top-level classes method definitions.
+        """
+        for node in ast.iter_child_nodes(self.tree):
+            if self._should_check_node(node):
+                yield node, False  # type: ignore
+
+            elif(
+                isinstance(node, ast.ClassDef)
```
Comment (mcproger) [inline flake8_fine_pytest/watchers/usefixtures.py:48]: Let's move it to helper, or at least to variable with understandable name.

File: tests/test_files/test_class_with_fixtures.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class TestClass:
```
Comment (mcproger) [inline tests/test_files/test_class_with_fixtures.py:1]: Why? We make flake-plugin for pytest, not for unittest. Also, we have an [issue](https://github.com/best-doctor/flake8-fine-pytest/issues/1) to forbid this type of syntax

File: flake8_fine_pytest/watchers/usefixtures.py (hunk 3 above)
Comment (mcproger) [reply flake8_fine_pytest/watchers/usefixtures.py:48]: Actually, we don't need this type of check

https://github.com/best-doctor/flake8-fine-pytest/pull/9/files#r533142561

File: tests/test_files/test_class_with_fixtures.py (hunk 4 above)
Comment (micheller) [reply tests/test_files/test_class_with_fixtures.py:1]: It's an open-source util, and pytest allows creating tests as class methods. Therefore, it has to be supported.

File: flake8_fine_pytest/watchers/usefixtures.py (hunk 1 above)
Comment (micheller) [reply flake8_fine_pytest/watchers/usefixtures.py:7]: yep

File: flake8_fine_pytest/watchers/usefixtures.py (hunk 2 above)
Comment (micheller) [reply flake8_fine_pytest/watchers/usefixtures.py:24]: not really, there's either an enable/disable option for a validator, or a integer/list options which defines validator's behavior.
So, in general that's not an should_run/should_not_run flag.

**Pull request merged**